python/mlc_llm/bench/executor.py

```diff
@@ -59,41 +49,89 @@ class FixedConcurrentRequestExecutor(Executor):
 
     def __init__(
         self,
-        api_endpoint: APIEndPoint,
+        f_create_api_endpoint: Callable[[], APIEndPoint],
+        num_processes: Optional[int],
         disable_tqdm: bool,
         num_concurrent_requests: int,
     ) -> None:
-        super().__init__(api_endpoint, disable_tqdm)
+        if num_processes is None:
+            # We assign each process at most 32 concurrent requests to send
+            # so that the asyncio pressure will not be too much.
+            num_processes = min((num_concurrent_requests + 31) // 32, 16)
```

Comment:
TODO in a followup, need to cap max number of processes to `multiprocessing.cpu_count() - ncores_needed_by_engine`